### PR TITLE
better determine when to plan optional+computed

### PR DESCRIPTION
**Pull request by jbardin**
In the common cases we cannot determine if a null optional+computed value in the configuration was previously set when the value also exists in the state. There is however one situation where we can determine if there was previously a configuration value, we can check if a complex object in state must have at least partially come from configuration by seeing if the prior value has any non-null attributes which are not computed in the schema.

### CODE REVIEW

**Comment by jbardin** [comment]
Actually let me see if I can extract all of that refactoring out of here, since it ends up being more related to the next PR.

File: internal/plans/objchange/objchange_test.go [hunk 1]
```diff
@@ -2036,14 +2036,14 @@ func TestProposedNew(t *testing.T) {
 				)),
 			}),
 			cty.ObjectVal(map[string]cty.Value{
-				"list_obj": cty.ListVal([]cty.Value{
-					cty.ObjectVal(map[string]cty.Value{
-						"obj": cty.ObjectVal(map[string]cty.Value{
-							"optional": cty.StringVal("prior"),
-							"computed": cty.StringVal("prior computed"),
+				"list_obj": cty.NullVal(cty.List(
```
Comment (kmoe) [inline internal/plans/objchange/objchange_test.go:2039]: Should the comment on ll. 1992-1993 be removed?
"The entire prior nested value should be represented in the proposed new object if the configuration is null."

**Review by kmoe — APPROVED** [review]
Nice!

File: internal/plans/objchange/objchange_test.go (hunk 1 above)
Comment (jbardin) [reply internal/plans/objchange/objchange_test.go:2039]: ooh, that's a good catch! The result is good, let me check that it's what we actually want at this point in the stack of PRs though (this was broken up from a single huge PR).

Comment (jbardin) [reply internal/plans/objchange/objchange_test.go:2040]: ok, this is the test I thought its was! updated the comment to describe the more accurate planning result.

**Pull request merged**